kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala

```diff
@@ -101,7 +101,6 @@ class KubernetesApplicationOperation extends ApplicationOperation with Logging {
           .inform(new SparkEnginePodEventHandler(kubernetesInfo))
         info(s"[$kubernetesInfo] Start Kubernetes Client Informer.")
         val engineSvcInformer = client.services()
-          .withLabelSelector(LABEL_KYUUBI_UNIQUE_KEY)
```

Comment:
this did not work as expect.

cc @zwangsheng do you have some suggestions?

Reply:
I wonder `withLabelSelector` can not work for this use case.
```
spec:
  selector:
    kyuubi-unique-tag: bf5fa281-0b2f-4096-aa0a-13a463c147a6
    spark-app-selector: spark-bf074093a7994954a89101eb2831bd1e
```

It is not label, just selector.

Reply:
Let us follow up in another PR.

Reply:
how about we seting kyuubi unique tag with `spark.kubernetes.driver.service.annotation.`, then we can match spark app with service annotation.

Reply:
<img width="1241" alt="image" src="https://github.com/apache/kyuubi/assets/6757692/ee376d75-053f-41da-8c46-6829b789ccc4">

Seems no filter api with annotation.

Reply:
I see. It's best if we can set kyuubi-unique-tag in svc label, but here is not related config.